api2-war/src/main/java/eu/europeana/api2/v2/service/translate/BeanTranslateService.java

```diff
@@ -240,13 +243,14 @@ boolean hasStaticTranslations(Proxy aggregatorProxy, Field field) {
         if (lang == null && !map.keySet().isEmpty()) {
             // return any value if available, but only if it's a supported language
             for (String key : map.keySet()) {
-                if (Language.isSupported(key)) {
-                    return Collections.singletonList(new FieldValuesLanguageMap(key, fieldName, map.get(key)));
+                 String isoLang = TranslationUtils.getISOLanguage(key);
+                if (Language.isSupported(isoLang)) {
+                    return Collections.singletonList(new FieldValuesLanguageMap(isoLang, fieldName, map.get(key)));
                 } else {
                     LOG.debug("  Found value for field {} in unsupported language {}", fieldName, key);
                 }
             }
-        } else if (lang != null && map.containsKey(lang)) {
+        } else if (lang != null && (map.containsKey(lang) || map.keySet().stream().anyMatch(key -> key.startsWith(lang)))) {
```

Comment:
Are you sure the stream and startswith test is necessary? We store only the 2 letter code in our map, don't we?
I added some logging to check and at least for the unit tests we never have the language code with locale here.